tests/TestCase.php

```diff
@@ -7,4 +7,11 @@
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function instance($abstract, $instance)
+    {
+        $this->app->bind($abstract, fn () => $instance);
+
+        return $instance;
+    }
```

Comment:
ℹ️ The `instance` method doesn't swap the class for the mock when the former is resolved with some parameters for some reason, but using `bind` instead works